### DISCUSSION TITLE
Expose css/sass defaultLoaders in config options.

### DISCUSSION THREAD

**Discussion by Izhaki** [vercel/next.js#24141] [Ideas]
### Describe the feature you'd like to request

NextJS already exposes `defaultLoaders.babel`, which can be used as follows to allow transpilation of soures outside the baseDir (in Monorepos):

```js
module.exports = withBundleAnalyzer({
  webpack: (config, options) => {
    // To allow transpiling aliases
    config.module.rules.push({
      test: /\.+(js|jsx|mjs|ts|tsx)$/,
      use: options.defaultLoaders.babel,
      include: Object.values(aliases),
    });

    return config;
  },
});
```

Given the built-in support for CSS/SCSS in NextJS, full support for monorepo setup will require to allow importing css/scss (in addition to js/ts) files from directories outside the baseDir.

# The issue with current workaround

The current way [next-transpile-modules does it](https://github.com/martpie/next-transpile-modules/blob/a37eeaeea882d28c8bbe6d0b7d38aa7982efd013/src/next-transpile-modules.js#L247-L274), is horribly hacky (@martpie):

```js
        // Support CSS modules + global in node_modules
        // TODO ask Next.js maintainer to expose the css-loader via defaultLoaders
        const nextCssLoaders = config.module.rules.find((rule) => typeof rule.oneOf === 'object');

        // .module.css
        if (nextCssLoaders) {
          const nextCssLoader = nextCssLoaders.oneOf.find(
            (rule) => rule.sideEffects === false && regexEqual(rule.test, /\.module\.css$/)
          );

          const nextSassLoader = nextCssLoaders.oneOf.find(
            (rule) => rule.sideEffects === false && regexEqual(rule.test, /\.module\.(scss|sass)$/)
          );

          if (nextCssLoader) {
            nextCssLoader.issuer.or = nextCssLoader.issuer.and ? nextCssLoader.issuer.and.concat(matcher) : matcher;
            delete nextCssLoader.issuer.not;
            delete nextCssLoader.issuer.and;
          } else {
            console.warn('next-transpile-modules - could not find default CSS rule, CSS imports may not work');
          }

          if (nextSassLoader) {
            nextSassLoader.issuer.or = nextSassLoader.issuer.and ? nextSassLoader.issuer.and.concat(matcher) : matcher;
            delete nextSassLoader.issuer.not;
            delete nextSassLoader.issuer.and;
          } else {
            console.warn('next-transpile-modules - could not find default SASS rule, SASS imports may not work');
          }
        }
```

### Describe the solution you'd like

```
options.defaultLoaders.css
options.defaultLoaders.sass
```

### Describe alternatives you've considered

- next-transpile-modules (doesn't worked for me with latest version; regardless, this request will solidify the plugin).
- https://github.com/vercel/next.js/pull/22867 - No support for css/sass yet.

**Comment by martpie** [comment]
Oh hell yes! 👍 

This would be extremely helpful for `next-transpile-modules` to be able to override/extend some of these rules. We don't need something state-of-the-art/sophisticated. It's fine if it breaks as well between releases. But just something to easily get these rules would already be extremely helpful.
